src/k8s/pkg/component/interface.go

```diff
@@ -0,0 +1,21 @@
+package component
+
+import "helm.sh/helm/v3/pkg/action"
+
+// HelmComponentManager defines an interface for managing k8s components.
+type HelmComponentManager interface {
```

Comment:
Let's keep the interface name generic. Only implementations should describe _how_ they implement the interface.
```suggestion
// ComponentManager defines an interface for managing k8s components.
type ComponentManager interface {
```